controller/src/main/java/org/jboss/as/controller/transform/description/BaseAttributeTransformationDescriptionBuilder.java

```diff
@@ -139,17 +151,34 @@ public interface BaseAttributeTransformationDescriptionBuilder<T extends BaseAtt
      * @param attributeName the attribute's original name
      * @param newName the new name for the attribute
      * @return this builder
+     *
+     * @deprecated use {@link #addRename(AttributeDefinition, AttributeDefinition)}
      */
+    @Deprecated(forRemoval = true)
     T addRename(AttributeDefinition attributeName, String newName);
 
+    /**
+     * Rename an attribute. An attribute can only be renamed once.
+     * <p>
+     * Renaming is done in the {@code RENAME} phase.
+     *
+     * @param oldAttribute the attribute's original definition
+     * @param newAttribute the new definition for the attribute
+     * @return this builder
+     */
+    T addRename(AttributeDefinition oldAttribute, AttributeDefinition newAttribute);
```

Comment:
NOTE: I considered adding a default impl for this new method. But it would have had to call the deprecated method. So later we would have to remove that default impl, thus breaking any theoretical implementing class using the default. And there is no such clase. The only impl class is the one I updated in this PR.